pkg/lso/localvolumediscovery_test.go

```diff
@@ -0,0 +1,404 @@
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testLocalVolumeDiscovery.Object)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryUpdate(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            string
+		testNodeSelector         corev1.NodeSelector
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            "",
+			testNodeSelector: corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+					MatchExpressions: []corev1.NodeSelectorRequirement{{
+						Key:      "cluster.ocs.openshift.io/openshift-storage",
+						Operator: "In",
+						Values:   []string{""},
+					}}},
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Nil(t, nil, testCase.testLocalVolumeDiscovery.Object)
+		testCase.testLocalVolumeDiscovery.WithNodeSelector(testCase.testNodeSelector)
+		_, err := testCase.testLocalVolumeDiscovery.Update()
+
+		if testCase.expectedError != "" {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
+			assert.Equal(t, testCase.testNodeSelector, *testCase.testLocalVolumeDiscovery.Definition.Spec.NodeSelector)
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryWithNodeSelector(t *testing.T) {
+	testCases := []struct {
+		testNodeSelector  corev1.NodeSelector
+		expectedError     bool
+		expectedErrorText string
+	}{
+		{
+			testNodeSelector: corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+					MatchExpressions: []corev1.NodeSelectorRequirement{{
+						Key:      "cluster.ocs.openshift.io/openshift-storage",
+						Operator: "In",
+						Values:   []string{""},
+					}}},
+				},
+			},
+			expectedError:     false,
+			expectedErrorText: "",
+		},
+		{
+			testNodeSelector: corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+					MatchExpressions: []corev1.NodeSelectorRequirement{{
+						Key:      "cluster.ocs.openshift.io/openshift-storage",
+						Operator: "Exists",
+					}}},
+				},
+			},
+			expectedError:     false,
+			expectedErrorText: "",
+		},
+		{
+			testNodeSelector: corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+					MatchExpressions: []corev1.NodeSelectorRequirement{{
+						Key:      "cluster.ocs.openshift.io/openshift-storage",
+						Operator: "Exists",
+					}, {
+						Key:      "machineconfiguration.openshift.io/role",
+						Operator: "In",
+						Values:   []string{"customcnf", "worker"},
+					}}},
+				},
+			},
+			expectedError:     false,
+			expectedErrorText: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidLVDObjectBuilder(buildLVDClientWithDummyObject())
+
+		result := testBuilder.WithNodeSelector(testCase.testNodeSelector)
+
+		if testCase.expectedError {
```

Comment:
Check error unconditionally, you can just use one variable for the expected error

Reply:
@trewest sameproblem on getting error msg